pymchelper/readers/topas.py

```diff
@@ -0,0 +1,232 @@
+from pathlib import Path
+import re
+from typing import List, Tuple, Union
+
+import numpy as np
+from pymchelper.axis import MeshAxis
+from pymchelper.estimator import Estimator
+from pymchelper.page import Page
+from pymchelper.readers.common import Reader, ReaderFactory
+
+
+class TopasReaderFactory(ReaderFactory):
+    """Factory for TopasReader"""
+
+    def get_reader(self):
+        """Return TopasReader if the file extension is .csv"""
+        if self.filename.endswith('.csv'):
+            return TopasReader
+        return None
+
+
+class TopasReader(Reader):
+    """Reader for Topas output files"""
+
+    def __init__(self, filename):
+        super(TopasReader, self).__init__(filename)
+        self.directory = Path(filename).parent
+
+    @staticmethod
+    def get_parameter_filename(results_data: str) -> str:
```

Comment:
consider migrating this to a generator approach, 
move this function outside of class
name it like `extract_parameter_filename`
apply it on a line, instead of full file content